crates/ethereum-forks/src/forkcondition.rs

```diff
@@ -85,9 +85,17 @@ impl ForkCondition {
     /// - The condition is satisfied by the timestamp;
     /// - or the condition is satisfied by the total difficulty
     pub fn active_at_head(&self, head: &Head) -> bool {
-        self.active_at_block(head.number) ||
-            self.active_at_timestamp(head.timestamp) ||
-            self.active_at_ttd(head.total_difficulty, head.difficulty)
+        match self {
+            Self::TTD { fork_block: Some(_), .. } => {
+                self.active_at_block(head.number) &&
+                    self.active_at_ttd(head.total_difficulty, head.difficulty)
+            }
+            _ => {
+                self.active_at_block(head.number) ||
+                    self.active_at_timestamp(head.timestamp) ||
+                    self.active_at_ttd(head.total_difficulty, head.difficulty)
+            }
+        }
```

Comment:
Here we would need a confirmation but here is the logic behind my modification:
- If we have a `TTD` with a block number `Some` then we know both the block number and the total difficulty. So for the `active_at_head` to be true both conditions must be validated and not just one.

For example if I have:

```rust
let head = Head {
    hash: Default::default(),
    number: 10,
    timestamp: 12345,
    total_difficulty: U256::from(1000),
    difficulty: U256::from(100),
};
```

Then the following fork condition:
```rust
let fork_condition =
ForkCondition::TTD { fork_block: Some(10), total_difficulty: U256::from(9000) };
```

should not be marked as `active_at_head` only because the block number is valid because the total difficulty is not valid. So for me, both elements must be validated. What do you think?

Reply:
@mattsse Can you confirm or not this is something considered as a bug? because I think that this impacts other tests where both block number and difficulty are not settled properly I imagine so that if this is a bug we probably need to modify these other tests too :)

Reply:
Fixed